include/dynd/kernels/multidispatch.hpp

```diff
@@ -349,6 +349,29 @@ namespace nd {
     template <int N0, int N1>
     struct new_multidispatch_by_type_id_kernel
         : base_virtual_kernel<new_multidispatch_by_type_id_kernel<N0, N1>> {
+      struct data {
```

Comment:
From reading this, I think the i[2] is to allow multi-dispatching based on any two of the source types, or the dst + a source type, is that right? Might be worth a small comment to help ease reading the code.